src/frontend/sklearn.cc

```diff
@@ -0,0 +1,272 @@
+/*!
+ * Copyright (c) 2021 by Contributors
+ * \file sklearn.cc
+ * \brief Frontend for scikit-learn models
+ * \author Hyunsu Cho
+ */
+#include <treelite/frontend.h>
+#include <treelite/tree.h>
+#include <memory>
+#include <queue>
+#include <algorithm>
+#include <numeric>
+#include <tuple>
+
+namespace treelite {
+namespace frontend {
+
+template <typename MetaHandlerFunc, typename LeafHandlerFunc>
+std::unique_ptr<treelite::Model> LoadSKLearnModel(
+    int n_trees, int n_features, int n_classes, const int64_t* node_count,
+    const int64_t** children_left, const int64_t** children_right, const int64_t** feature,
+    const double** threshold, const double** value, const int64_t** n_node_samples,
+    const double** impurity, MetaHandlerFunc meta_handler, LeafHandlerFunc leaf_handler) {
+  CHECK_GT(n_trees, 0);
+  CHECK_GT(n_features, 0);
+
+  std::unique_ptr<treelite::Model> model_ptr = treelite::Model::Create<double, double>();
+  meta_handler(model_ptr.get(), n_features, n_classes);
+  auto* model = dynamic_cast<treelite::ModelImpl<double, double>*>(model_ptr.get());
+
+  for (int tree_id = 0; tree_id < n_trees; ++tree_id) {
```

Comment:
Optional: consider OpenMP parallelization. This is also fine to do in a separate pull request.

Reply:
Sure. Will do in a future pull request.